openssl/src/x509/mod.rs

```diff
@@ -1431,6 +1431,26 @@ impl X509ReqBuilder {
         }
     }
 
+    /// Adds attribute defined by name and type to certificate request
+    pub fn add_attribute(
+        &mut self,
+        attrname: &str,
+        attrtype: Asn1Type,
+        value: &[u8],
+    ) -> Result<(), ErrorStack> {
+        unsafe {
+            let attrname = CString::new(attrname).unwrap();
+            cvt(ffi::X509_REQ_add1_attr_by_txt(
+                self.0.as_ptr(),
+                attrname.as_ptr(),
+                attrtype.as_raw(),
+                value.as_ptr(),
+                value.len() as c_int,
```

Comment:
Please use `value.len().try_into().expect("value's len outside range")` so that there's no silent truncation.